runelite-client/src/main/java/net/runelite/client/plugins/friendnotes/FriendNotesPlugin.java

```diff
@@ -84,19 +100,94 @@
 	@Inject
 	private ChatboxPanelManager chatboxPanelManager;
 
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private FriendNotesConfig config;
+
 	@Getter
 	private HoveredFriend hoveredFriend = null;
 
+	private int iconIdx = -1;
+	private String currentlyLayouting = null;
+
+	@Provides
+	private FriendNotesConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(FriendNotesConfig.class);
+	}
+
 	@Override
 	protected void startUp() throws Exception
 	{
 		overlayManager.add(overlay);
+		loadIcon();
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			rebuildFriendsList();
+			rebuildIgnoreList();
+		}
+	}
+
+	@Subscribe
```

Comment:
While this file doesn't seem to maintain the usual definitions order of `getConfig()`, `startUp()`, `shutDown()`, `@Subscribe`rs, then instance/static methods (it has some instance methods mixed with the subscribers), `shutDown()` should definitely be defined here before defining any more subscribers.